webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -762,46 +761,47 @@ class ProjectsResponderADM(responderData: ResponderData) extends Responder(respo
             }
 
             // check if the supplied shortname is unique
-            shortnameExists <- projectByShortnameExists(createRequest.shortname)
+            shortnameExists <- projectByShortnameExists(createProjectRequest.shortname)
             _ = if (shortnameExists) {
-                throw DuplicateValueException(s"Project with the shortname: '${createRequest.shortname}' already exists")
+                throw DuplicateValueException(s"Project with the shortname: '${createProjectRequest.shortname}' already exists")
             }
 
             validatedShortcode = StringFormatter.getGeneralInstance.validateProjectShortcode(
-                createRequest.shortcode,
-                errorFun = throw BadRequestException(s"The supplied short code: '${createRequest.shortcode}' is not valid.")
+                createProjectRequest.shortcode,
+                errorFun = throw BadRequestException(s"The supplied short code: '${createProjectRequest.shortcode}' is not valid.")
             )
 
             // check if the optionally supplied shortcode is valid and unique
             shortcodeExists <- {
                 projectByShortcodeExists(validatedShortcode)
             }
             _ = if (shortcodeExists) {
-                throw DuplicateValueException(s"Project with the shortcode: '${createRequest.shortcode}' already exists")
+                throw DuplicateValueException(s"Project with the shortcode: '${createProjectRequest.shortcode}' already exists")
+            }
+            newProjectIRI: IRI = createProjectRequest.projectIri match {
+                case Some(customProjectIri) => customProjectIri.toString
```

Comment:
Where do you validate the custom IRI? It needs to be a valid project IRI, or you will not be able to retrieve the project.

Reply:
and you don't need the `.toString` since it is already a string.